### ISSUE TITLE
"Only ebooks" filter on author pages excludes "Preview only" books

### ISSUE THREAD

**Issue by el4ctr0n** [internetarchive/openlibrary#8888]
### Problem

To reproduce the bug, go to this URL and see this book:
https://openlibrary.org/authors/OL6708339A/John_D._MacDonald?mode=everything&sort=rating
![Screenshot_20240308-232533_Chrome](https://github.com/internetarchive/openlibrary/assets/19307323/b43b602c-ffc7-4869-afc1-9f6964bb1c05)

When you use the "only ebooks" filter on the same author age,  this "Preview only" book isn't shown:
https://openlibrary.org/authors/OL6708339A/John_D._MacDonald?mode=ebooks&sort=rating

### Reproducing the bug

* Expected behavior: "Preview only" books are not hidden when the "only ebooks" filter is used on author pages.
* Actual behavior: "Preview only" books are hidden when the "only ebooks" filter is used on author pages.

### Context

- Browser (Chrome, Safari, Firefox, etc): Chrome
- OS (Windows, Mac, etc): Android
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

**Comment by mekarpeles** [comment]
I'd love to know what your use case is / why you may want preview-only books to show up under "ebooks"? This would help us evaluate.

**Comment by el4ctr0n** [comment]
> I'd love to know what your use case is / why you may want preview-only books to show up under "ebooks"? This would help us evaluate.

Preview-only books still contain information from the book's contents, even though I'd need to look elsewhere for the whole thing.
In situations where there are numerous books on an author page (or, for that matter, a search results page) there should be a way to filter out completely undigitized books (with not even a preview). Currently, filtering for ebooks means preview-only books are treated like undigitized books, and I have to forgo filtering for ebooks if I want to see preview-only ebooks along with full-access ebooks.

**Comment by el4ctr0n** [comment]
> I'd love to know what your use case is / why you may want preview-only books to show up under "ebooks"? This would help us evaluate.

@mekarpeles  
Preview-only books still contain information from the book's contents, even though I'd need to look elsewhere for the whole thing.
In situations where there are numerous books on an author page (or, for that matter, a search results page) there should be a way to filter out completely undigitized books (with not even a preview). Currently, filtering for ebooks means preview-only books are treated like undigitized books, and I have to completely forgo filtering for ebooks if I want to see preview-only ebooks along with full-access ebooks.

If not as a default feature, perhaps you could add a user settings option to include preview-only books when filtering author pages and search results pages for ebooks.

**Comment by el4ctr0n** [comment]
Any update on this issue? Thanks in advance.

**Comment by el4ctr0n** [comment]
Can there please be a status update on this issue?

**Comment by el4ctr0n** [comment]
> Note: in the meantime you can use this search as a workaround: `ebook_access:[printdisabled TO *]

Thank you so much for this workaround.
Is there also a search operator for searching exclusively for Preview-only books?

**Comment by el4ctr0n** [comment]
> Note: in the meantime you can use this search as a workaround: `ebook_access:[printdisabled TO *]

@cdrini 
Thank you so much for this workaround.
Is there also a search operator for searching exclusively for Preview-only books?

**Comment by mekarpeles** [comment]
> we don't have a solution for this yet

It used to be the case that anything that had been digitized, whether borrowable or not, would show up when the `ebooks` radio button was selected.

Especially as more items may become unavailable for policy reasons, I think it makes sense to switch back to showing any item that's been digitized (especially to researchers who may wish to search inside or do Table of Contents matching) as this seem more valuable to patrons than hiding previews and requiring them to filter by `everything`.

As mentioned, a future step could make this a dropdown, and maybe we want an issue broken down for this, e.g.:
* Readable Books
* Books with Previews
* Entire Catalog

**Comment by el4ctr0n** [comment]
> As mentioned, a future step could make this a dropdown, and maybe we want an issue broken down for this, e.g.:
> 
> * Readable Books
> * Books with Previews
> * Entire Catalog

@mekarpeles 
Can there please also be a filtering option for digitized but removed books like [https://github.com/internetarchive/openlibrary/issues/8888#issuecomment-2454822375](this one)?

**Comment by el4ctr0n** [comment]
> As mentioned, a future step could make this a dropdown, and maybe we want an issue broken down for this, e.g.:
> 
> * Readable Books
> * Books with Previews
> * Entire Catalog

@mekarpeles 
Can there please also be a filtering option for digitized but removed books like [this one](https://openlibrary.org/books/OL1704131M/From_noon_to_starry_night)?

**Comment by cdrini** [comment]
@mekarpeles I think adding that drop down would be great if someone wants to add it! It would be similar to the sort dropdown on the search page; so this would require designs.

@el4ctr0n Here are searches for [preview only](https://openlibrary.org/search?q=ebook_access%3Aprintdisabled&mode=everything) or [removed books](https://openlibrary.org/search?q=from+noon+to+starry+night+ia%3A*+-ebook_access%3A%5Bprintdisabled+TO+*%5D&mode=everything). These might not be _perfectly_ accurate, but should be close :+1:

**Comment by el4ctr0n** [comment]
> @el4ctr0n Here are searches for [preview only](https://openlibrary.org/search?q=ebook_access%3Aprintdisabled&mode=everything) or [removed works](https://openlibrary.org/search?q=from+noon+to+starry+night+ia%3A*+-ebook_access%3A%5Bprintdisabled+TO+*%5D&mode=everything). These might not be _perfectly_ accurate, but should be close 👍

@cdrini 
Thank you so much for these searches.
Is there also a search operator that I could use to search for openlibrary books without cover images?

**Comment by cdrini** [comment]
Yep! You can add `-cover_i:*` to your search, eg https://openlibrary.org/search?q=sherlock+holmes+-cover_i%3A*&mode=everything

Note this will find works which don't have any edition with a cover.

**Comment by el4ctr0n** [comment]
@cdrini Is there also a search operator to hide from the openlibrary search results books in Japanese?
Additionally, is there a search operator to search for openlibrary books without any external edition or work identifiers (no ISBN, no Worldcat, no Wikidata)?

**Comment by cdrini** [comment]
For the first case, you can do `-language:jpn` . Note this will exclude works which have _any_ edition in Japanese. If you only want to avoid japanese editions, but still want to see results that have editions in other languages, I think you might be able to do `-edition.language:jpn`, but not 100% on that one!

For the second case, you'd need to explicitly list out the identifiers you care about, eg `-isbn:* -id_wikidata:* -oclc:*`

**Comment by el4ctr0n** [comment]
@cdrini Thank you so much for these search operators.
Can you also please tell me the search operator for searching openlibrary.org for books with a specific place of publication (for example, books that were published in Monaco)?

**Comment by cdrini** [comment]
Oh, apparently we do index that; `publish_place` :P But not it's only aggregated at the work level, so you won't be able to search for _which_ edition was published in that place, you'll just know that the work has at least one edition that was published in eg Monaco. Also note there isn't any smart processing of this field to normalize spelling, language, format, country, etc, so it'll be a bit of a grab bag in terms of how well it works.

eg https://openlibrary.org/search?q=publish_place:Monaco
